src/lightning_app/core/api.py

```diff
@@ -146,6 +161,42 @@ async def get_spec(
     return app_spec or []
 
 
+@fastapi_service.post("/api/v1/commands", response_class=JSONResponse)
+async def post_command(
+    request: Request,
+) -> None:
+    data = await request.json()
+    command_name = data.get("command_name", None)
+    if not command_name:
+        raise Exception("The provided command name is empty.")
+    command_arguments = data.get("command_arguments", None)
+    if not command_arguments:
+        raise Exception("The provided command metadata is empty.")
+    affiliation = data.get("affiliation", None)
+    if not affiliation:
+        raise Exception("The provided affiliation is empty.")
+
+    async def fn(request: Request):
+        data = await request.json()
+        request_id = data["id"]
+        api_commands_requests_queue.put(data)
+
+        resp = api_commands_responses_queue.get()
+        if request_id == resp["id"]:
+            return resp["response"]
+
+        raise Exception("This is a bug. It shouldn't happen.")
```

Comment:
I think it's very possible this happens if we have multiple command requests to different flows.

Reply:
Yes, 100 %. Let me add a TODO.

Reply:
I think this needs to be fixed structurally. Managing requests/response in two separate queues de facto limits us to one single request/response pair at any time. There may be commands that take different amounts of time to execute, which would lead to ordering to not be respected.

So we either:
* address this issue structurally now (by changing the logic to not use queues this way)
* we lock the command API in case there's a command in flight. Users will get an error like "a command is currently executing" (pretty ugly but at least correct)
* we ensure that request/response queues strictly have the same ordering, but this is very tricky to guarantee in the face of errors 

Am I getting this right?

Reply:
I have a global store that can hold responses in case of asyncio executes the thread back in a different order. I didn't fully connect it. I will add this to the PR so we are covered. I was being a bit lazy.

To answer Luca's points, in order to scale the number of requests, we would need to execute them within a thread possibly. But right now, I believe sequential execution provides more value overall.